class/Patchwork/Utf8.php

```diff
@@ -555,7 +555,7 @@ static function substr_replace($s, $replace, $start, $len = 2147483647)
     static function ucfirst($s)
     {
         $c = iconv_substr($s, 0, 1, 'UTF-8');
-        return static::ucwords($c) . substr($s, strlen($c));
+        return static::strtoupper($c) . substr($s, strlen($c));
```

Comment:
You didn't read the ICU page :) The DZ ligature e.g. exists as a single UTF-8 char.
This one has 3 char cases:
- lower, which is a single `dz` char
- upper, which is a single `DZ` char
- title, which is a single `Dz` char

As you see, lcfirst must use strtolower, and ucfirst the title case version